### PR TITLE
dialog: create component

### PR DESCRIPTION
**Pull request by GarriGarrison**

### CODE REVIEW

File: src/components/hive-dialog/hive-dialog.vue [hunk 1]
```diff
@@ -0,0 +1,140 @@
+<script lang="ts" setup>
+import { CommonProps } from '@/common/mixin/props';
+import { Update } from '@/common/mixin/emits';
+
+export interface HiveDialogProps extends CommonProps {
```
Comment (AntonL9vov) [inline src/components/hive-dialog/hive-dialog.vue]: In other components they were just Props.

File: src/components/hive-dialog/hive-dialog.vue [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+  modelValue: boolean;
+  top?: string;
+  left?: string;
+  maskBackground?: string;
+}
+
+withDefaults(defineProps<HiveDialogProps>(), {
+  modelValue: false,
+  top: '40%',
+  left: '40%',
```
Comment (AntonL9vov) [inline src/components/hive-dialog/hive-dialog.vue]: Strange 40%

File: src/components/hive-dialog/hive-dialog.vue [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+  maskBackground: '#262d34ad',
+});
+
+const emit = defineEmits<Update<boolean>>();
```
Comment (AntonL9vov) [inline src/components/hive-dialog/hive-dialog.vue]: In other components emit types were separeted

File: src/components/hive-dialog/hive-dialog.vue [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+
+const handleHide = () => {
+
+  emit('update:modelValue', false);
```
Comment (AntonL9vov) [inline src/components/hive-dialog/hive-dialog.vue]: Use the function from emits.ts 
(btw function from emits.ts needs to be fixed, change "value: Ref<T>" to "value: T") 
UPD: fix done in next pull request

File: src/components/hive-dialog/hive-dialog.vue [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+    <transition name="backdrop">
+      <div v-if="modelValue" class="hive-dialog__fade">
+        <div class="hive-dialog__mask" @click="handleHide" :style="{ background: maskBackground }" />
+      </div>
+    </transition>
+
+    <transition name="content">
+      <div v-if="modelValue" ref="content" class="hive-dialog" :style="{ top: top, left: left }">
+        <button class="hive-dialog__btn-close" @click="handleHide">
+          <img class="hive-dialog__btn-close-img" src="./icons/remove.svg" alt="Notification close button" />
```
Comment (AntonL9vov) [inline src/components/hive-dialog/hive-dialog.vue]: Absolute path would be better

File: src/components/hive-dialog/hive-dialog.vue (hunk 1 above)
Comment (AntonL9vov) [reply src/components/hive-dialog/hive-dialog.vue]: Still HiveDialogProps...

File: src/components/hive-dialog/hive-dialog.vue (hunk 3 above)
Comment (AntonL9vov) [reply src/components/hive-dialog/hive-dialog.vue]: Still not separeted...

File: src/components/hive-dialog/hive-dialog.vue (hunk 4 above)
Comment (AntonL9vov) [reply src/components/hive-dialog/hive-dialog.vue]: Still not the function...

File: src/components/hive-dialog/hive-dialog.vue [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+  background-color: $dialog-bg;
+  border-radius: $border-radius;
+  min-width: $dialog-width-min;
+  min-height: $dialog-height-min;
+  max-width: fit-content;
+  max-height: fit-content;
+  left: calc(-50vw + 50%);
+  right: calc(-50vw + 50%);
+  top: calc(-50vh + 50%);
+  bottom: calc(-50vh + 50%);
```
Comment (AntonL9vov) [inline src/components/hive-dialog/hive-dialog.vue:66]: Is there any to reason to add left and right (top and bottom) at the same time

**Review by AntonL9vov — APPROVED** [review]
LGTM

**Review by anastansa — APPROVED** [review]

**Pull request merged**